synapse/handlers/receipts.py

```diff
@@ -147,23 +148,67 @@ async def received_client_receipt(
             receipt_type=receipt_type,
             user_id=user_id,
             event_ids=[event_id],
-            data={"ts": int(self.clock.time_msec())},
+            data={"ts": int(self.clock.time_msec()), "hidden": hidden},
         )
 
         is_new = await self._handle_new_receipts([receipt])
         if not is_new:
             return
 
-        if self.federation_sender:
+        if self.federation_sender and (
+            not hidden or not self.hs.config.experimental.msc2285_enabled
+        ):
```

Comment:
I think it would be better to use something like:

```suggestion
        if self.federation_sender and not (
            self.hs.config.experimental.msc2285_enabled and hidden
        ):
```

Otherwise it makes the condition a bit too difficult to read.